### PR TITLE
[engine] Align story execution timeout properties naming

### LINKED ISSUE #3416: Align story timeout properties naming
I think it should be moved to `Story` as well

_Originally posted by @valfirst in https://github.com/vividus-framework/vividus/pull/3414#discussion_r1037510217_

### PR DESCRIPTION
**Pull request by ikalinin1**
fixes #3416

### CODE REVIEW

File: vividus-engine/src/main/java/org/vividus/batch/BatchStorage.java [hunk 1]
```diff
@@ -40,7 +45,17 @@ public BatchStorage(IPropertyMapper propertyMapper, String defaultStoryExecution
             List<String> defaultMetaFilters, boolean failFast) throws IOException
     {
         this.defaultMetaFilters = defaultMetaFilters;
-        this.defaultStoryExecutionTimeout = Duration.ofSeconds(Long.parseLong(defaultStoryExecutionTimeout));
+        if (NumberUtils.isCreatable(defaultStoryExecutionTimeout))
+        {
+            this.defaultStoryExecutionTimeout = Duration.ofSeconds(Long.parseLong(defaultStoryExecutionTimeout));
+            LOGGER.warn(
+                "Usage of plain seconds for `story.execution-timeout` is deprecated and will be removed in"
+                    + " VIVIDUS 0.7.0. Please use Duration instead.");
```
Comment (valfirst) [inline vividus-engine/src/main/java/org/vividus/batch/BatchStorage.java:53]: I think new property shouldn't allow to set duration using old format

File: vividus-engine/src/main/java/org/vividus/batch/BatchConfiguration.java [hunk 2]
```diff
@@ -167,19 +166,25 @@ private List<String> convertToList(String list)
 
     private static final class StoryExecutionConfiguration
     {
-        private boolean failFast;
+        private Boolean failFast;
+        private Duration executionTimeout;
 
-        private void setFailFast(boolean failFast)
+        private void setFailFast(Boolean failFast)
```
Comment (uarlouski) [inline vividus-engine/src/main/java/org/vividus/batch/BatchConfiguration.java:172]: why is it requred to change to boxed type?

Comment (ikalinin1) [reply vividus-engine/src/main/java/org/vividus/batch/BatchConfiguration.java:172]: To Keep previous behaviour

File: vividus-engine/src/main/java/org/vividus/batch/BatchStorage.java [hunk 3]
```diff
@@ -24,23 +24,38 @@
...
-            List<String> defaultMetaFilters, boolean failFast) throws IOException
+    public BatchStorage(IPropertyMapper propertyMapper, Duration defaultStoryExecutionTimeout,
+        String deprecatedDefaultStoryExecutionTimeout, List<String> defaultMetaFilters,
+            boolean failFast) throws IOException
     {
         this.defaultMetaFilters = defaultMetaFilters;
-        this.defaultStoryExecutionTimeout = Duration.ofSeconds(Long.parseLong(defaultStoryExecutionTimeout));
+        if (null != deprecatedDefaultStoryExecutionTimeout)
+        {
+            this.defaultStoryExecutionTimeout =
```
Comment (valfirst) [inline vividus-engine/src/main/java/org/vividus/batch/BatchStorage.java:49]: I think we need to fail if both deprecated and non-deprecated properties are specified

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-engine/src/main/java/org/vividus/batch/BatchStorage.java [hunk 4]
```diff
@@ -24,23 +24,44 @@
...
+                "Conflicting properties are found: `bdd.story-execution-timeout` and `story.execution-timeout`. "
+                    + DEPRECATION_MESSAGE);
+            this.defaultStoryExecutionTimeout =
+                Duration.ofSeconds(Long.parseLong(deprecatedDefaultStoryExecutionTimeout));
+            LOGGER.warn(DEPRECATION_MESSAGE);
+        }
+        else
+        {
+            this.defaultStoryExecutionTimeout = defaultStoryExecutionTimeout == null ? DEFAULT_STORY_TIMEOUT
+                : defaultStoryExecutionTimeout;
```
Comment (valfirst) [inline vividus-engine/src/main/java/org/vividus/batch/BatchStorage.java:63]: optional comment:
```suggestion
            this.defaultStoryExecutionTimeout = Optional.ofNullable(defaultStoryExecutionTimeout).orElse(DEFAULT_STORY_TIMEOUT);
```

**Review by valfirst — APPROVED** [review]

**Review by uarlouski — APPROVED** [review]

**Pull request merged**